scripts/main.js

```diff
@@ -0,0 +1,337 @@
+let statusBar = document.getElementById("statusBar");
+let gameTitle = document.getElementById("game-title");
+let cells = Array.from(document.getElementsByClassName("cell"));
+
+//will store our current game here, empty array helps to track the played cells.
+let boxes = ["", "", "", "", "", "", "", "", ""];
+let gameStatus = true;
+let totalTurns = 0;
+
+// Initialization of player variables
+let currentPlayer = "X";
+let player1 = document.getElementById("player1");
+let player2 = document.getElementById("player2");
+let playerVsPlayer = document.getElementById("playerVPlayer");
+let playerVsComputer = document.getElementById("playerVComputer");
+let gameType = "";
+
+
+//Clock timer Initialization
+let hours = document.getElementById("hours");
+let minutes = document.getElementById("minutes");
+let seconds = document.getElementById("seconds");
+let totalSeconds = 0;
+
+//----Check pVp or pVComp-----//
+// start is disabled until player selects the game type:pVp or pVcomp
+start.disabled = true;
+reset.disabled = true;
+playerVsComputer.addEventListener("click", () => {
+  player1.disabled = true;
+  player2.disabled = true;
+  playerVsPlayer.disabled = true;
+  playerVsComputer.disabled = true;
+  start.disabled = false;
+  reset.disabled = false;
+  gameType = "pVComp";
+  currentPlayer = "X";
+  statusBar.textContent = "Click Start Button";
+});
+
+playerVsPlayer.addEventListener("click", () => {
+  
+  playerVsPlayer.disabled = true;
+  playerVsComputer.disabled = true;
+  start.disabled = false;
+  reset.disabled = false;
+  gameType = "pVp";
+  currentPlayer = "X";
+  statusBar.textContent = "Click Start Button";
+});
+
+//-----Function to decide players turn---->
+//----Start button function----//
+//Start button fading out and displaying player
+
+start.addEventListener("click", startGame);
+
+function startGame() {
+  start.disabled = true;
+  totalTurns = 0;
+  // disables the button once clicked
+  statusBar.textContent = "Enjoy the Game";
+  gameTimer = setInterval(setTime, 1000, true);
+  if (gameType === "pVp") {
+    playerGame()
+  } else if (gameType === "pVComp") {
+    computerGame()
+  }
+}
+
+//---Restart button function----//
+//restarts the game
+reset.addEventListener("click", resetGame);
+
+function resetGame() {
+  start.disabled = false;
+  playerVsPlayer.disabled = false;
+  playerVsComputer.disabled = false;
+  boxes = ["", "", "", "", "", "", "", "", ""];
+  cells = document.querySelectorAll(".cell").forEach((cell) => (cell.innerHTML = ""));
+  currentPlayer = "X";
+  //gameTimer = setInterval(setTime, 1000, true);
+  //statusBar.textContent = player1.value + "'s Turn";
+  c0.style.backgroundColor = "";
+  c1.style.backgroundColor = "";
+  c2.style.backgroundColor = "";
+  c3.style.backgroundColor = "";
+  c4.style.backgroundColor = "";
+  c5.style.backgroundColor = "";
+  c6.style.backgroundColor = "";
+  c7.style.backgroundColor = "";
+  c8.style.backgroundColor = "";
+  setTime(false);
+}
```

Comment:
I believe the switching game mode issue is coming from somewhere in this block. Steps to recreate problem:
- play a game mode until completion
- switch game modes to the opposite mode
- try to start and play the game (should break here)

If doing this with the inspector tool open in the browser window you can see something is happening with the cells variable at that point. It looks like changing around the reset logic or adding some logic could resolve this issue.